src/include/distributed/pg_dist_shard.h

```diff
@@ -43,13 +42,12 @@ typedef FormData_pg_dist_shard *Form_pg_dist_shard;
  *      compiler constants for pg_dist_shards
  * ----------------
  */
-#define Natts_pg_dist_shard 6
```

Comment:
@marcocitus are these changes incompatible if the pg_dist_shard table retains the shardalias column? It seems that they are incompatible. So how about I do the following:

1. Preserve the presence of shardalias column in citus.sql and in relevant regression tests.
2. Revert the changes in pg_dist_shard.h, but  change Anum_pg_dist_shard_shardalias to Anum_pg_dist_shard_shardalias_DEPRECATED.
3. Restore the code in master_metadata_utility.c that sets null on shardalias in the constructed tuple.

Does that sound right?